### ISSUE TITLE
Task7

### ISSUE THREAD

**Issue by faramer86** [faramer86/BIOPYTHON_ITMO_2018#7]
Done!

**Comment by YakutovDmitriy** [comment]
* `valuesunion` - ok
* `popcount` - ok
* `powers` - ok
* `subpalindrome` - you doesn't take into account lexicographical order. Fix it
* `isIPv4` - try something like this `isIPv4('asdfasdfdsa')`
* `fibonacci` - cool!

**Comment by faramer86** [comment]
Done!

**Comment by YakutovDmitriy** [comment]
* `subpalindrome`. You use a cache but it is always empty because you create a new one after each function calling. Please fix it
* `isIPv4`. Try this: `+49.77.30.104`

**Comment by faramer86** [comment]
Almost Done! Is it appropriate to use permutation approach in `bruckets2`?

**Comment by YakutovDmitriy** [comment]
* `subpalindrome`. Now you doesn't consider lexicographical order of strings. Fix it
* `bruckets2`. Do you mean `brackets2`?
* Other functions are ok

**Comment by faramer86** [comment]
Yes, I mean `bruckets2`. I just want to write a function that generate lots of permutations and cut off all inappropriate.

**Comment by YakutovDmitriy** [comment]
Still `bruckets2`? Please rename the function in correct way. Then I'll check it

**Comment by faramer86** [comment]
Done!

**Comment by YakutovDmitriy** [comment]
* `subpalindrome` - ok
* `brackets2` - this functions returns nothing. It should yield values
